File: CS/ReplaceDefaultActionInListView.Module.Win/EditAddressBookRecordController.cs
using System;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.ExpressApp.SystemModule;
using DevExpress.Persistent.Base;

namespace ReplaceDefaultActionInListView.Module.Win {
    public class EditAddressBookRecordController : ViewController<ListView> {
        public EditAddressBookRecordController() {
            TargetObjectType = typeof(AddressBookRecord);
            SimpleAction editAddressBookRecordAction = 
                new SimpleAction(this, "EditAddressBookRecord", PredefinedCategory.Edit);
            editAddressBookRecordAction.ImageName = "Action_Edit";
            editAddressBookRecordAction.SelectionDependencyType = 
                SelectionDependencyType.RequireSingleObject;
            editAddressBookRecordAction.Execute += editAddressBookRecordAction_Execute;
        }
        void editAddressBookRecordAction_Execute(object sender, SimpleActionExecuteEventArgs e) {
            ListViewProcessCurrentObjectController.ShowObject(
                e.CurrentObject, e.ShowViewParameters, Application, Frame, View);
        }
    }
}

File: VB/ReplaceDefaultActionInListView.Module.Win/EditAddressBookRecordController.vb
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.ExpressApp.SystemModule
Imports DevExpress.Persistent.Base

Namespace ReplaceDefaultActionInListView.Module.Win
    Public Class EditAddressBookRecordController
        Inherits ViewController(Of ListView)

        Public Sub New()
            TargetObjectType = GetType(AddressBookRecord)
            Dim editAddressBookRecordAction As New SimpleAction(Me, "EditAddressBookRecord", PredefinedCategory.Edit)
            editAddressBookRecordAction.ImageName = "Action_Edit"
            editAddressBookRecordAction.SelectionDependencyType = SelectionDependencyType.RequireSingleObject
            AddHandler editAddressBookRecordAction.Execute, AddressOf editAddressBookRecordAction_Execute
        End Sub
        Private Sub editAddressBookRecordAction_Execute(ByVal sender As Object, ByVal e As SimpleActionExecuteEventArgs)
            ListViewProcessCurrentObjectController.ShowObject(e.CurrentObject, e.ShowViewParameters, Application, Frame, View)
        End Sub
    End Class
End Namespace
